batchflow/models/tf/encoder_decoder.py

```diff
@@ -169,12 +169,12 @@ def head(cls, inputs, targets, name='head', **kwargs):
         kwargs = cls.fill_params('head', **kwargs)
 
         with tf.variable_scope(name):
-            x = cls.crop(inputs, targets, kwargs['data_format'])
-            x = super().head(x, name, **kwargs)
+            x = super().head(inputs, name, **kwargs)
+            x = cls.crop(x, targets, kwargs['data_format'])
```

Comment:
Lets make a check: if spatial shape of `inputs` is bigger than that of `targets`, we `crop`. Then we call `head`, then we `crop` once again